File: 3018/3018.go
func maximumNumberOfRemovalQueriesT(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := maximumNumberOfRemovalQueriesT(root.Left)
	r := maximumNumberOfRemovalQueriesT(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 3018/3018.cpp
class Solution {
public:
    int maximumNumberOfRemovalQueriesT(TreeNode* root) {
        if(!root) return 0;
        return 1+max(maximumNumberOfRemovalQueriesT(root->left),maximumNumberOfRemovalQueriesT(root->right));
    }
};